lib/inc/pxp-agent/util/windows/pid_file.hpp

```diff
@@ -0,0 +1,70 @@
+#ifndef SRC_AGENT_UTIL_WINDOWS_PID_FILE_HPP_
+#define SRC_AGENT_UTIL_WINDOWS_PID_FILE_HPP_
+
+#include <string>
+#include <stdexcept>
+#include <boost/filesystem/path.hpp>
+using HANDLE = void *;
+using DWORD = unsigned long;
+
+namespace PXPAgent {
+namespace Util {
+
+class PIDFile {
+  public:
+    struct Error : public std::runtime_error {
+        explicit Error(std::string const& msg) : std::runtime_error(msg) {}
+    };
+
+    static const std::string FILENAME;
+
+    // Create the PID file directory if necessary.
+    // Open the PID file.
+    // Throw a PIDFile::Error if dir_path is not a directory or in
+    // case of a failure when opening the PID file.
+    // NB: we pass dir_path to enable testing; DIR_PATH is hardcoded
+    // in Configuration
+    PIDFile(const std::string& dir_path);
+
+    // Perform clean up unless already done.
+    ~PIDFile();
+
+    // Disable copies, and provide move semantics.
+    PIDFile(PIDFile const&) = delete;
+    PIDFile& operator=(PIDFile const&) = delete;
+    PIDFile(PIDFile &&) = default;
+    PIDFile& operator=(PIDFile &&) = default;
+
+    // Perform clean up; use logging if log is true
+    // logging is disabled in the destructor, as we can't guarantee
+    // it'll be destroyed before Boost.Log is shut down
```

Comment:
This only happens because I implemented cleanup differently. I'm using global static destruction on Windows, where I think you circumvent that on Linux. I'll double check.